cdhweb/static_src/global/rich-text.scss

```diff
@@ -1,56 +1,95 @@
-// Styles for body copy, aka "rich text".
-// Wrap body copy elements in a div/section with the class "rich-text".
-// Works best if the direct children are the typographic elements
-// (headings, p etc). This ensures that the .rich-text container doesn't
-// have extra space at the start/end.
-// :where() selector is used to keep specificity low.
-:where(.rich-text) {
-  --space-between-els: 1lh; // relative so it scales to bigger/smaller text
-  --space-between-sections: #{px2rem(40)};
-
-  @include xl {
-    --space-between-sections: #{px2rem(64)};
-  }
-
-  > * {
-    margin-bottom: var(--space-between-els, 1em);
-  }
-
-  // When we encounter a heading, that can be considered a new "section" of
-  // the content so should have a nice big space above...
-  // Note: h1s are assumed to be outside of the `rich-text` area, as they're
-  // typically in a hero banner rather than coming from a WYSIWYG with other text.
-  :is(h2, h3, h4, h5, h6) {
-    margin-top: var(--space-between-sections);
-  }
-  // ...unless a heading appears immediately after another heading, in which case we need to reduce the spacing.
-  h2 + h3,
-  h3 + h4,
-  h4 + h5,
-  h5 + h6 {
-    margin-top: var(--space-between-els, 1em);
-  }
-
-  // Remove any orphan vertical space at the start/end of the rich-text block.
-  // NOTE: this assumes items are direct children of .rich-text container.
-  > * {
-    &:first-child {
-      margin-top: 0;
-    }
-    &:last-child {
-      margin-bottom: 0;
-    }
-  }
-
-  ul,
-  ol {
-    // Undo padding:0 from the css reset, which kills list indentation.
-    padding: revert;
-  }
-
-  // Collapsing margins should ensure we don't get extra space between this and preceding element.
-  // i.e. no `:not(:first...)` required.
-  li {
-    margin-top: 0.5em;
+// By default, every prose element has a bottom margin.
+// `li` is excluded because list item spacing is handled later on.
+// Same with headings because they have their own, more specific spacing values.
+// Ignore header and footer, they have their own styles.
+:where(main) {
+  --space-default: 1rem;
+
+  --space-l2-lg: #{px2rem(56)};
+  --space-l3-lg: #{px2rem(40)};
+  --space-l4-lg: #{px2rem(24)};
+  --space-l5-lg: #{px2rem(40)};
+  --space-l6-lg: #{px2rem(40)};
+
+  --space-l1-sm: calc(var(--space-l1-lg) / 4);
```

Comment:
are you missing a `--space-l1-lg` declaration?

Reply:
ah, more that i removed it and forgot to remove the `sm` version. will remove this too.

l1 (i.e. `h1`) is only ever in the hero area, so we don't need longform text styles for it.

Reply:
might be nice to have a comment above these variables explaining that `l` is equivalent to heading-level

Reply:
True, i will add a comment I guess. The reason I didn't name them e.g. h3-space is because they get applied to the element _before_ the h3 and it felt odd(er) to say that a `p` had the `h3` space.

Reply:
yep, totally get that. Maybe 'equivalent' isn't the right word — it's more to let devs know that this isn't some kind of 8px modular spacing thing where you could keep adding `l7`, `l8`, etc, but that there's a relationship to the heading levels. It also might help explain why variables for two different levels can have the same spacing.

Reply:
(and why there's no `l1`)